Runtimes/Core/stubs/CMakeLists.txt

```diff
@@ -28,6 +28,7 @@ endif()
 
 target_compile_definitions(swiftStdlibStubs PRIVATE
   $<$<BOOL:${BUILD_SHARED_LIBS}>:-DswiftCore_EXPORTS>
+  $<$<NOT:$<BOOL:${BUILD_SHARED_LIBS}>>:-DSWIFT_STATIC_STDLIB>
```

Comment:
Since the `SWIFT_STATIC_STDLIB` is part of the `swiftShims` interface, let's make this definition of the `swiftShims` target interface definitions so that you only need to set it there instead of on all of the things pulling in `swiftShims`.

Reply:
Hmm, let me see if that is sufficient. I'd tried to do that, but found that it didn't percolate through everything.

Reply:
Hmm, anything that "links" the swift shims privately won't forward it. Technically, things that are including the shims that don't explicitly link it (or have it passed transitively) should fail to find the headers though.

Reply:
I don't think that its sufficient - the stubs don't pull in the shims for example. So you do end up with this needing to be applied at the various sites. We _can_ push it globally, but I'm not sure if that is a better approach.

Reply:
The stubs do depend on the shims though. https://github.com/swiftlang/swift/blob/8f6ad4c5cdae8f904f22abf3245740f295fb7b49/Runtimes/Core/stubs/CMakeLists.txt#L33-L34